### PR TITLE
fix duplication of dijkstra.cpp in greedy_algorithms

### PR DESCRIPTION
**Pull request by xgupta**
#### Description of Change

File dijkstra.cpp  is placed in two directory graph and greedy_algorithms which cause a problem in building with CMake.  This change makes it possible to build the dijkstra.cpp in greedy_algorithms.

#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes: 

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/C-Plus-Plus/pull/1259"></a>

### CODE REVIEW

**Review by Panquesito7 — APPROVED** [review]

**Comment by xgupta** [comment]
I think we should commit the change to upstream.

File: greedy_algorithms/dijkstra_spf.cpp [hunk 1]
```diff
@@ -33,7 +32,7 @@ class Graph {
     }
 };
 // Utility function to find minimum distance vertex in mdist
-int minDistance(int mdist[], bool vset[], int V) {
+int minDistance(std::vector<int> mdist, std::vector<bool> vset, int V) {
```
Comment (kvedala) [inline greedy_algorithms/dijkstra_spf.cpp:35]:
```suggestion
int minDistance(const std::vector<int> &mdist, const std::vector<bool> &vset) {
uint32_t V = mdist.size();
```

File: greedy_algorithms/dijkstra_spf.cpp [hunk 2]
```diff
@@ -46,13 +45,13 @@ int minDistance(int mdist[], bool vset[], int V) {
 }
 
 // Utility function to print distances
-void print(int dist[], int V) {
-    cout << "\nVertex  Distance" << endl;
+void print(std::vector<int> dist, int V) {
```
Comment (kvedala) [inline greedy_algorithms/dijkstra_spf.cpp:48]:
```suggestion
void print(const std::vector<int> &dist) {
uint32_t V = dist.size();
```

File: greedy_algorithms/dijkstra_spf.cpp [hunk 3]
```diff
@@ -1,7 +1,6 @@
-#include <limits.h>
+#include <climits>
 #include <iostream>
-
-using namespace std;
+#include <vector>
 
 // Wrapper class for storing a graph
```
Comment (kvedala) [inline greedy_algorithms/dijkstra_spf.cpp:5]:
```suggestion

namespace greedy_algorithms {

namespace dijkstra {

// Wrapper class for storing a graph
```

File: greedy_algorithms/dijkstra_spf.cpp [hunk 4]
```diff
@@ -94,31 +93,32 @@ void Dijkstra(Graph graph, int src) {
 int main() {
```
Comment (kvedala) [inline greedy_algorithms/dijkstra_spf.cpp:93]:
```suggestion
} // namespace dijkstra
} // namespace greedy_algorithms

// Driver function
int main() {
```

File: greedy_algorithms/dijkstra_spf.cpp [hunk 5]
```diff
@@ -61,8 +60,8 @@ void print(int dist[], int V) {
 // weights
 void Dijkstra(Graph graph, int src) {
```
Comment (kvedala) [inline greedy_algorithms/dijkstra_spf.cpp:61]:
```suggestion
void Dijkstra(const Graph &graph, int src) {
```

File: greedy_algorithms/dijkstra_spf.cpp [hunk 6]
```diff
@@ -94,31 +93,32 @@ void Dijkstra(Graph graph, int src) {
...
             i--;
             continue;
         }
     }
-    cout << "\nEnter source:";
-    cin >> gsrc;
+    std::cout << "\nEnter source:";
+    std::cin >> gsrc;
     Dijkstra(G, gsrc);
+    delete[] G.edges;
```
Comment (kvedala) [inline greedy_algorithms/dijkstra_spf.cpp:121]: This delete is incorrect. 
Please add a destructor to the Graph class that will deallocate the memory.

**Review by kvedala — CHANGES_REQUESTED** [review]

**Comment by kvedala** [comment]
This is a good opportunity to fix some of the older incorrect code. 😄

**Pull request closed without merging**